core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -86,22 +86,23 @@ oppia.factory('responsesService', [
             .can_have_solution && stateSolutionService.savedMemento !== null) {
           if (stateSolutionService.savedMemento.correctAnswer !== null) {
             var currentStateName = editorContextService.getActiveStateName();
-            SolutionHelperService.verifySolution(
+            SolutionVerificationService.verifySolution(
               explorationContextService.getExplorationId(),
               explorationStatesService.getState(currentStateName),
               stateSolutionService.savedMemento.correctAnswer,
               function() {
                 explorationStatesService.updateSolutionValidity(
-                  currentStateName, false);
+                  currentStateName, true);
                 alertsService.addInfoMessage(
-                  'That solution does not lead to the next state!');
+                  'The solution is now valid!');
```

Comment:
Not sure I get it, sorry. Say you add responses then you add a correct solution. Then you change the responses such that the solution is still correct (e.g. you fix a typo in one of the wrong item selection responses). What happens?